.buildkite/serve.rayci.yml

```diff
@@ -24,9 +25,27 @@ steps:
       - bazel run //ci/ray_ci:test_in_docker -- //python/ray/serve/... //python/ray/tests/...  serve 
         --except-tags post_wheel_build,gpu,worker-container,xcommit
         --workers "$${BUILDKITE_PARALLEL_JOB_COUNT}" --worker-id "$${BUILDKITE_PARALLEL_JOB}" --parallelism-per-worker 3
+        --test-env=EXPECTED_PYTHON_VERSION=3.8
     depends_on: servepydantic2build
     job_env: forge
 
+  - label: ":ray-serve: serve: Python 3.11 tests"
+    parallelism: 2
+    # Run only on postmerge or if the PR includes the `python-3-11-support` label.
+    if: build.env("BUILDKITE_PIPELINE_ID") == "0189e759-8c96-4302-b6b5-b4274406bf89" || build.pull_request.labels includes "python-3-11-support"
+    tags:
+      - serve
+      - python
+    instance_type: large
+    soft_fail: true
+    commands:
+      - bazel run //ci/ray_ci:test_in_docker -- //python/ray/serve/... //python/ray/tests/...  serve
+        --except-tags post_wheel_build,gpu,worker-container,xcommit
+        --workers "$${BUILDKITE_PARALLEL_JOB_COUNT}" --worker-id "$${BUILDKITE_PARALLEL_JOB}" --parallelism-per-worker 3
```

Comment:
need to add the flag `--build-name servepython311build` and it will work (by default it will pick servebuild)